### PR TITLE
Update Icon component and docs

### PR DESCRIPTION
**Pull request by daileytj**
Fixes BLUI-4370.

#### Changes proposed in this Pull Request:
- Add/update icon component and docs to work with MD3

#### Screenshots / Screen Recording (if applicable)
Simulator Screenshot - iPhone 14 - 2023-08-15 at 15 12 32

#### To Test:
- `cd md3`
- `yarn && cd ios && pod install && cd ..`
- `yarn ios` or `yarn android`
- observe various forms of icons rendered in channel values
- ensure all potential use cases are covered in the docs and examples

update: you can now test this in the showcase on branch `feature/blui-4370-icon-component`

### CODE REVIEW

**Comment by huayunh** [comment]
- The screenshot shows the icon and the text is not aligned properly. Is that a Channel Value issue or an icon issue?

**Comment by daileytj** [comment]
> * The screenshot shows the icon and the text is not aligned properly. Is that a Channel Value issue or an icon issue?

Looks like a channelvalue issue to me. 
Screenshot 2023-08-16 at 8 11 52 AM

**Comment by huayunh** [comment]
And do our BLUI icons work with RNP's [IconButton component](https://callstack.github.io/react-native-paper/docs/components/IconButton/)? Sorry if this is a dumb question... I was trying to figure out how our icons are injected but didn't see the docs for that.

**Comment by daileytj** [comment]
> And do our BLUI icons work with RNP's [IconButton component](https://callstack.github.io/react-native-paper/docs/components/IconButton/)? Sorry if this is a dumb question... I was trying to figure out how our icons are injected but didn't see the docs for that.

Not a dumb question. I was wondering the same thing. Based on the documentation it doesn't look like we can use a custom icon for icon button, but I'll have to dig deeper into that. @joebochill do you have any insights around this?

**Comment by joebochill** [comment]
There are several ways to customize the icon for iconButton (or any other component that takes an icon, presumably):

image
image

The quickest way to do it is probably to use the function: `icon={(iconProps) => <OurIcon {...iconProps} />}`

You could also use a SettingsProvider to set the icon library to use with the `string` version of the prop, but this is a little hairier since it doesn't let you mix libraries, so you'd have multiple providers all over to switch between BLUI and MD icons:
image

File: components/src/md3/ChannelValue/ChannelValue.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    return (
+        <View style={[defaultStyles.root, styles.root, style]} {...viewProps}>
+            {getIcon()}
+            <Text
+                variant={'bodyMedium'}
+                numberOfLines={1}
+                ellipsizeMode={'tail'}
+                testID={'text-wrapper'}
+                // fontSize={fontSize}
+                style={[{ color: getColor(), fontSize: fontSize, fontFamily: 'OpenSans-Regular' }]}
```
Comment (ektaghag-eaton) [inline components/src/md3/ChannelValue/ChannelValue.tsx:177]: Is this a temporary solution? are we going to update `bodyMedium`  in the theme file?

Comment (daileytj) [reply components/src/md3/ChannelValue/ChannelValue.tsx:177]: Yes, the font-family will be coming from the variant. I only put the channelvalue in here as a way to test the Icon utility. There is another story to update the ChannelValue for MD3

**Review by ektaghag-eaton — APPROVED** [review]

**Pull request merged**